system/lib/pthread/pthread_create.c

```diff
@@ -248,8 +248,18 @@ int __pthread_create(pthread_t* restrict res,
   // want libc.need_locks to be set from the moment it starts.
   if (!libc.threads_minus_1++) libc.need_locks = 1;
 
+  // Assign the pthread_t object over immediately, so that by the time pthread_create_js()
+  // is dispatched to a pthread and the pthread main runs, the value will be visible to
+  // the thread to examine.
```

Comment:
lgtm, but I wonder if you understand why exactly writing `res` fixes the issue?

__pthread_create_js rakes the `new` pointer where was `res` is basically a pointer to a pointer which is owned by the calling thread.   Do you know why is it important the `res` be set during `__pthread_create_js`?

Reply:
Thanks for the clarification.  Sorry I didn't see that.